### PR TITLE
Namespace

### PR DESCRIPTION
**Pull request by FlaminGuy**
Adding shills for each function.

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/phase/jvm/GenNamespaces.scala [hunk 1]
```diff
@@ -76,6 +84,106 @@ object GenNamespaces {
   }
 
   /**
+    * Adding a shill for the function `defn` on namespace `ns`
+    */
+  private def compileShillMethod(visitor: ClassWriter,
+                                 defn: Def,
+                                 sym: Symbol.DefnSym,
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/jvm/GenNamespaces.scala]: We don't need the symbol when we have the def.

File: main/src/ca/uwaterloo/flix/language/phase/jvm/GenNamespaces.scala [hunk 2]
```diff
@@ -76,6 +84,106 @@ object GenNamespaces {
...
+                                 nsClassType: JvmType.Reference,
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/jvm/GenNamespaces.scala]: We can also get the namespace class type from the namespace info

File: main/src/ca/uwaterloo/flix/language/phase/jvm/GenNamespaces.scala [hunk 3]
```diff
@@ -76,6 +84,106 @@ object GenNamespaces {
...
+                                 ifoType: JvmType.Reference,
+                                 ns: NamespaceInfo)(implicit root: Root, flix: Flix): Unit = {
+    // Name of the shill
+    val name = JvmOps.getDefMethodNameInNamespaceClass(sym)
+
+    // Jvm type of method args
+    val args = defn.tpe.typeArguments.map(JvmOps.getErasedType)
+
+    // Length of args in local
+    val varLen = args.init.map{
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/jvm/GenNamespaces.scala]: We need some better name. Perhaps stack height or stack size etc?

File: main/src/ca/uwaterloo/flix/language/phase/jvm/GenNamespaces.scala [hunk 4]
```diff
@@ -76,6 +84,106 @@ object GenNamespaces {
...
+      val iLoad = AsmOps.getLoadInstruction(arg)
+      method.visitVarInsn(iLoad, offset)
+
+      // Call the setter for the argument
+      method.visitMethodInsn(INVOKEVIRTUAL, ifoType.name.toInternalName, s"setArg$index",
+        AsmOps.getMethodDescriptor(List(arg), JvmType.Void), false)
+
+      // Incrementing the offset
+      arg match {
+        case JvmType.PrimLong | JvmType.PrimDouble => offset += 2
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/jvm/GenNamespaces.scala]: Lets refactor this into a helper function in AsmOps or JvmOps.

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/jvm/GenNamespaces.scala]: Perhaps `getStackSpace` or similar.

File: main/src/ca/uwaterloo/flix/language/phase/jvm/GenNamespaces.scala [hunk 5]
```diff
@@ -76,6 +84,106 @@ object GenNamespaces {
...
+        case _ => offset += 1
+      }
+    }
+
+    // TODO: Replace this with an appropriate while loop, for now we just call the apply
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/jvm/GenNamespaces.scala]: Can we do that before merging in, or not?

Comment (FlaminGuy) [reply main/src/ca/uwaterloo/flix/language/phase/jvm/GenNamespaces.scala]: I rather wait until we want to do `apply` expressions in `compileExp`.

**Pull request merged**